redwood-tooling-schema/src/test/kotlin/app/cash/redwood/tooling/schema/SchemaParserTest.kt

```diff
@@ -31,18 +31,27 @@ import assertk.assertions.hasMessage
 import assertk.assertions.isEmpty
 import assertk.assertions.isEqualTo
 import assertk.assertions.isNull
+import com.google.testing.junit.testparameterinjector.TestParameter
+import com.google.testing.junit.testparameterinjector.TestParameterInjector
 import example.redwood.ExampleSchema
+import java.io.File
 import kotlin.DeprecationLevel.HIDDEN
+import kotlin.reflect.KClass
+import org.junit.Assume.assumeTrue
 import org.junit.Test
+import org.junit.runner.RunWith
 
-class SchemaParserTest {
+@RunWith(TestParameterInjector::class)
+class SchemaParserTest(
+  @TestParameter private val parser: SchemaParser,
+) {
   object TestScope
 
   interface NonAnnotationSchema
 
   @Test fun nonAnnotatedSchemaThrows() {
     assertFailsWith<IllegalArgumentException> {
-      parseProtocolSchemaSet(NonAnnotationSchema::class)
+      parser.parse(NonAnnotationSchema::class)
     }.hasMessage(
       "Schema app.cash.redwood.tooling.schema.SchemaParserTest.NonAnnotationSchema missing @Schema annotation",
     )
```

Comment:
This is the only test that's working in FIR. It means we can successfully look up the schema type and can notice that it's missing annotations.

Reply:
👍🏻